modules/parquet/src/parquetjs/classes/parquet-cursor.ts

```diff
@@ -0,0 +1,95 @@
+// Forked from https://github.com/kbajalc/parquets under MIT license (Copyright (c) 2017 ironSource Ltd.)
+import {FileMetaData} from '../parquet-thrift';
+import {ParquetEnvelopeReader} from './parquet-envelope-reader';
+import {ParquetSchema} from '../schema/schema';
+import {ParquetRecord} from '../schema/declare';
+import * as Shred from '../schema/shred';
+
+/**
+ * A parquet cursor is used to retrieve rows from a parquet file in order
+ */
+export class ParquetCursor<T> implements AsyncIterable<T> {
+  public metadata: FileMetaData;
+  public envelopeReader: ParquetEnvelopeReader;
+  public schema: ParquetSchema;
+  public columnList: string[][];
+  public rowGroup: ParquetRecord[];
+  public rowGroupIndex: number;
+
+  /**
+   * Create a new parquet reader from the file metadata and an envelope reader.
+   * It is usually not recommended to call this constructor directly except for
+   * advanced and internal use cases. Consider using getCursor() on the
+   * ParquetReader instead
+   */
+  constructor(
+    metadata: FileMetaData,
+    envelopeReader: ParquetEnvelopeReader,
+    schema: ParquetSchema,
+    columnList: string[][]
+  ) {
+    this.metadata = metadata;
+    this.envelopeReader = envelopeReader;
+    this.schema = schema;
+    this.columnList = columnList;
+    this.rowGroup = [];
```

Comment:
Nit: The constant values can be initialized on line 16

Reply:
Done